notify/drivers/sfdc.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2016: Mirantis Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import json
+import logging
+from xml.dom import minidom
+
+import requests
+from requests.packages.urllib3 import exceptions as urllib_exc
+
+from notify import config
+from notify.drivers import base
+
+requests.packages.urllib3.disable_warnings(urllib_exc.InsecureRequestWarning)
+
+
+LOG = logging.getLogger("sfdc")
+LOG.setLevel(config.get_config().get("logging", {}).get("level", "INFO"))
+
+
+class OAuth2(object):
+
+    def __init__(self,
+                 client_id,
+                 client_secret,
+                 username,
+                 password,
+                 auth_url=None,
+                 organizationId=None):
+        self.auth_url = auth_url or "https://login.salesforce.com"
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.username = username
+        self.password = password
+        self.organization = organizationId
+
+    def authenticate_soap(self):
+        LOG.debug("Making SFDC SOAP auth for {}".format(self.username))
+        doc = minidom.Document()
+        env = doc.appendChild(doc.createElement("soapenv:Envelope"))
+        env.setAttribute("xmlns:soapenv",
+                         "http://schemas.xmlsoap.org/soap/envelope/")
+        env.setAttribute("xmlns:urn", "urn:partner.soap.sforce.com")
+        header = env.appendChild(doc.createElement("soapenv:Header"))
+        a = header.appendChild(doc.createElement("urn:CallOptions"))
```

Comment:
Remove please hardcoded make a loop iterate over the elemennts to add them properly.